core/src/main/scala/com/nvidia/spark/rapids/tool/Platform.scala

```diff
@@ -215,6 +215,13 @@ abstract class Platform(var gpuDevice: Option[GpuDevice],
   def defaultGpuDevice: GpuDevice = T4Gpu
   def defaultNumGpus: Int = 1
 
+  /**
+   * Fraction of the system’s total memory that is available for executor use.
+   * This value should be adjusted based on the platform to account for memory
+   * reserved by the resource managers (e.g., YARN).
+   */
+  def fractionOfAvailableMemory: Double = 1.0
```

Comment:
The name is confusing. AvailableMemory indicates free memory to be used, but actually this is used in a different way.